### ISSUE TITLE
azureblob : Received error: context deadline exceeded after 1m0s transfer

### ISSUE THREAD

**Issue by ccvalley** [ncw/rclone#2647]
#### What is the problem you are having with rclone?
Transfer of large files (via either sync or copy) times out and restarts transfer after every 1m0s. With the error "context deadline exceeded"

#### What is your rclone version (output from `rclone version`)
rclone v1.43-150-g249a523d-beta
- os/arch: darwin/amd64
- go version: go1.11.1

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
Mac OS X 10.11.6

####  Which cloud storage system are you using? (eg Google Drive)
Azure Blob Storage

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
rclone copy -v --stats=5s source dest

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)
```
~/gene-idx/data >> rclone copy -vv --stats=5s delphistor:gene-idx/sales sales
2018/10/12 14:06:54 DEBUG : rclone: Version "v1.43-150-g249a523d-beta" starting with parameters ["rclone" "copy" "-vv" "--stats=5s" "delphistor:gene-idx/sales" "sales"]
2018/10/12 14:06:54 DEBUG : Using config file from "/Users/ccvalley/.config/rclone/rclone.conf"
2018/10/12 14:06:54 DEBUG : normalized_sales.csv: Sizes differ (src 421979546 vs dst 23064576)
2018/10/12 14:06:54 DEBUG : sales_by_gene.csv: Sizes differ (src 211484449 vs dst 16773120)
2018/10/12 14:06:54 INFO  : Local file system at /Users/ccvalley/gene-idx/data/sales: Waiting for checks to finish
2018/10/12 14:06:54 DEBUG : prodcode_to_gene.csv: Size and modification time the same (differ by -229.605274ms, within tolerance 1s)
2018/10/12 14:06:54 DEBUG : prodcode_to_gene.csv: Unchanged skipping
2018/10/12 14:06:54 DEBUG : unique_genes.csv: Size and modification time the same (differ by -245.605156ms, within tolerance 1s)
2018/10/12 14:06:54 DEBUG : unique_genes.csv: Unchanged skipping
2018/10/12 14:06:54 INFO  : Local file system at /Users/ccvalley/gene-idx/data/sales: Waiting for transfers to finish
2018/10/12 14:06:59 INFO  :
Transferred:   	   21.992M / 604.118 MBytes, 4%, 4.124 MBytes/s, ETA 2m21s
Errors:                 0
Checks:                 4 / 4, 100%
Transferred:            0 / 2, 0%
Elapsed time:        5.3s
Transferring:
 *                             sales_by_gene.csv:  3% /201.687M, 1.248M/s, 2m35s
 *                          normalized_sales.csv:  3% /402.431M, 2.999M/s, 2m9s

2018/10/12 14:07:04 INFO  :
Transferred:   	   38.992M / 604.118 MBytes, 6%, 3.774 MBytes/s, ETA 2m29s
Errors:                 0
Checks:                 4 / 4, 100%
Transferred:            0 / 2, 0%
Elapsed time:       10.3s
Transferring:
 *                             sales_by_gene.csv:  6% /201.687M, 1.444M/s, 2m9s
 *                          normalized_sales.csv:  6% /402.431M, 2.666M/s, 2m21s

... [removed redundant stats lines] ...

2018/10/12 14:07:49 INFO  :
Transferred:   	  247.992M / 604.118 MBytes, 41%, 4.481 MBytes/s, ETA 1m19s
Errors:                 0
Checks:                 4 / 4, 100%
Transferred:            0 / 2, 0%
Elapsed time:       55.3s
Transferring:
 *                             sales_by_gene.csv: 52% /201.687M, 1.965M/s, 48s
 *                          normalized_sales.csv: 35% /402.431M, 2.516M/s, 1m43s

2018/10/12 14:07:54 INFO  :
Transferred:   	  266.992M / 604.118 MBytes, 44%, 4.425 MBytes/s, ETA 1m16s
Errors:                 0
Checks:                 4 / 4, 100%
Transferred:            0 / 2, 0%
Elapsed time:      1m0.3s
Transferring:
 *                             sales_by_gene.csv: 56% /201.687M, 1.871M/s, 46s
 *                          normalized_sales.csv: 38% /402.431M, 2.487M/s, 1m40s

2018/10/12 14:07:54 NOTICE: normalized_sales.csv: Removing partially written file on error: context deadline exceeded
2018/10/12 14:07:54 DEBUG : normalized_sales.csv: Received error: context deadline exceeded - low level retry 1/10
2018/10/12 14:07:54 NOTICE: sales_by_gene.csv: Removing partially written file on error: context deadline exceeded
2018/10/12 14:07:54 DEBUG : sales_by_gene.csv: Received error: context deadline exceeded - low level retry 1/10
2018/10/12 14:07:59 INFO  :
Transferred:   	  288.242M / 873.368 MBytes, 33%, 4.412 MBytes/s, ETA 2m12s
Errors:                 0
Checks:                 4 / 4, 100%
Transferred:            0 / 2, 0%
Elapsed time:      1m5.3s
Transferring:
 *                             sales_by_gene.csv:  4% /201.687M, 2.248M/s, 1m25s
 *                          normalized_sales.csv:  2% /402.431M, 1.999M/s, 3m16s
```

**Comment by ncw** [comment]
Rclone's idle timer is 5minues by default, not 1 minute so I'm guessing this is something internal to the Azure SDK.

The only 1 minute timers I see in the SDK source are here

https://github.com/ncw/rclone/blob/249a523dd3646d6933bcff19bdd633b3d0ba693f/vendor/github.com/Azure/azure-storage-blob-go/2018-03-28/azblob/zc_policy_retry.go#L97-L106

That looks like something we could override.

@sandeepkru can you replicate this?  Any ideas?

**Comment by ncw** [comment]
I tried to replicate this by copying a file with a bwlimit so it took longer than a minute, but it didn't show the problem.

    rclone --bwlimit 100k -vv --stats 1s copyto azure:rclone-test1/10M /tmp/10M.copy

**Comment by wpbrown** [comment]
I think the problem is clear when you get the error output I do. I'm not sure why the errors are so different for @ccvalley .

> Put https\://ZZZ.blob.core.windows.net/test/test.bin?comp=blocklist&**timeout=61**: context canceled

It's best not to use the timeout parameter. You can't increase the timeout beyond 30 seconds for all but the data transfer operations and by setting timeout on blob transfer operations you actually override the default rate based limit (10 minutes per megabyte for uploads). See [Timeouts for Blob Service](https://docs.microsoft.com/en-us/rest/api/storageservices/setting-timeouts-for-blob-service-operations).

Even though my transfer is almost certainly failing at 1 minutes, I don't get an error reported until almost 5 minutes. Also you'll notice the status output below is broken in multiple ways. It immediately and erroneously reports 100% has been transferred. It's actually transferring at about 1.4MiB/s according to iftop. It also doesn't show any errors have occurred even though the transfer is failing and retrying. 

I'm running 1.43.1 on Ubuntu 18.04.1. 

<details>
<summary>(click for output)$ rclone -vv --stats=10s copy test.bin azure_test:test/</summary>

<pre>
will@volta:~/dev$ rclone -vv --stats=10s copy test.bin azure_test:test/
2018/10/13 19:39:47 DEBUG : rclone: Version "v1.43.1" starting with parameters ["rclone" "-vv" "--stats=10s" "copy" "test.bin" "azure_test:test/"]
2018/10/13 19:39:47 DEBUG : Using config file from "/home/will/.config/rclone/rclone.conf"
2018/10/13 19:39:47 DEBUG : test.bin: Couldn't find file - need to transfer

2018/10/13 19:39:57 INFO  : 
Transferred:   	      220M / 220 MBytes, 100%, 21.973 MBytes/s, ETA 0s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:         10s
Transferring:
 *                                      test.bin: 100% /220M, 24.441M/s, 0s

2018/10/13 19:40:07 INFO  : 
Transferred:   	      220M / 220 MBytes, 100%, 10.993 MBytes/s, ETA 0s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:         20s
Transferring:
 *                                      test.bin: 100% /220M, 11.328M/s, 0s

2018/10/13 19:40:17 INFO  : 
Transferred:   	      220M / 220 MBytes, 100%, 7.330 MBytes/s, ETA 0s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:         30s
Transferring:
 *                                      test.bin: 100% /220M, 5.941M/s, 0s

2018/10/13 19:40:27 INFO  : 
Transferred:   	      220M / 220 MBytes, 100%, 5.498 MBytes/s, ETA 0s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:         40s
Transferring:
 *                                      test.bin: 100% /220M, 3.116M/s, 0s

[ ... redundant updates removed ... ]

2018/10/13 19:43:37 INFO  : 
Transferred:   	      220M / 220 MBytes, 100%, 979.426 kBytes/s, ETA 0s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:       3m50s
Transferring:
 *                                      test.bin: 100% /220M, 15/s, 0s

2018/10/13 19:43:47 INFO  : 
Transferred:   	      220M / 220 MBytes, 100%, 938.618 kBytes/s, ETA 0s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:        4m0s
Transferring:
 *                                      test.bin: 100% /220M, 8/s, 0s

2018/10/13 19:43:57 INFO  : 
Transferred:   	      220M / 220 MBytes, 100%, 901.075 kBytes/s, ETA 0s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:       4m10s
Transferring:
 *                                      test.bin: 100% /220M, 4/s, 0s

2018/10/13 19:44:07 INFO  : 
Transferred:   	      220M / 220 MBytes, 100%, 866.420 kBytes/s, ETA 0s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:       4m20s
Transferring:
 *                                      test.bin: 100% /220M, 2/s, 0s

2018/10/13 19:44:17 INFO  : 
Transferred:   	      220M / 220 MBytes, 100%, 834.332 kBytes/s, ETA 0s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:       4m30s
Transferring:
 *                                      test.bin: 100% /220M, 1/s, 0s

2018/10/13 19:44:27 INFO  : 
Transferred:   	      220M / 220 MBytes, 100%, 804.536 kBytes/s, ETA 0s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:       4m40s
Transferring:
 *                                      test.bin: 100% /220M, 0/s, 0s

2018/10/13 19:44:35 ERROR : test.bin: Failed to copy: -> github.com/ncw/rclone/vendor/github.com/Azure/azure-pipeline-go/pipeline.newDefaultHTTPClientFactory.func1.1, /home/travis/gopath/src/github.com/ncw/rclone/vendor/github.com/Azure/azure-pipeline-go/pipeline/core.go:234
HTTP request failed

Put https://ZZZ.blob.core.windows.net/test/test.bin?comp=blocklist&timeout=61: context canceled
2018/10/13 19:44:35 ERROR : Attempt 1/3 failed with 1 errors and: -> github.com/ncw/rclone/vendor/github.com/Azure/azure-pipeline-go/pipeline.newDefaultHTTPClientFactory.func1.1, /home/travis/gopath/src/github.com/ncw/rclone/vendor/github.com/Azure/azure-pipeline-go/pipeline/core.go:234
HTTP request failed

Put https://ZZZ.blob.core.windows.net/test/test.bin?comp=blocklist&timeout=61: context canceled
2018/10/13 19:44:35 DEBUG : test.bin: Couldn't find file - need to transfer
2018/10/13 19:44:37 INFO  : 
Transferred:   	      220M / 440 MBytes, 50%, 776.794 kBytes/s, ETA 4m50s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:       4m50s
Transferring:
 *                                      test.bin:  0% /220M, 0/s, -

2018/10/13 19:44:47 INFO  : 
Transferred:   	      440M / 440 MBytes, 100%, 1.467 MBytes/s, ETA 0s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:        5m0s
Transferring:
 *                                      test.bin: 100% /220M, 19.997M/s, 0s

2018/10/13 19:44:57 INFO  : 
Transferred:   	      440M / 440 MBytes, 100%, 1.419 MBytes/s, ETA 0s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:       5m10s
Transferring:
 *                                      test.bin: 100% /220M, 9.956M/s, 0s

2018/10/13 19:45:07 INFO  : 
Transferred:   	      440M / 440 MBytes, 100%, 1.375 MBytes/s, ETA 0s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:       5m20s
Transferring:
 *                                      test.bin: 100% /220M, 5.222M/s, 0s

2018/10/13 19:45:17 INFO  : 
Transferred:   	      440M / 440 MBytes, 100%, 1.333 MBytes/s, ETA 0s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:       5m30s
Transferring:
 *                                      test.bin: 100% /220M, 2.739M/s, 0s

^C
will@volta:~/dev$ 
</pre>

</details>

**Comment by wpbrown** [comment]
I noticed in my log above the error was for `comp=blocklist`. But I know in the past that I've seen the error with `comp=block`. Neither of which makes sense because my blob is 220MiB, < 256MiB chunking cutoff. It looks like `azblob.UploadStreamToBlockBlob` used for the non-chunk rclone backend is always using the chunking REST API (see the methods of `uploadStreamToBlockBlobOptions`). This would still work, but its still including that timeout=61 even for the block PUTs. You might not notice this if you use the default 4MiB chunk size. However, I had set the chunk size very large, so with the 1min timeout, even the chunk puts are presumably timing out.

With chunksize 100MiB:
```
BufferSize:      int(o.fs.opt.ChunkSize) + 1, // +1 Needed until https://github.com/Azure/azure-storage-blob-go/pull/75 is merged
```
Also the +1 here is causing the valid chunksize config of 100MiB to fail:

```
===== RESPONSE ERROR (ServiceCode=RequestBodyTooLarge) =====
Description=The request body is too large and exceeds the maximum permissible limit.
RequestId:5d959ea2-101e-013b-68e9-6391bf000000
Time:2018-10-14T18:15:16.2425545Z, Details: 
   Code: RequestBodyTooLarge
   MaxLimit: 104857600
   PUT https://XXX.blob.core.windows.net/test/test.bin?blockid=qkxnTsOcTjxCS6f1LQKlmAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA%3D%3D&comp=block&timeout=61
   Authorization: REDACTED
   Content-Length: [104857601]
   User-Agent: [Azure-Storage/0.1 (go1.11.1; linux)]
   X-Ms-Client-Request-Id: [b836bff9-97ee-4239-56ab-df850cc21f84]
   X-Ms-Date: [Sun, 14 Oct 2018 18:15:16 GMT]
   X-Ms-Version: [2018-03-28]
   --------------------------------------------------------------------------------
   RESPONSE Status: 413 The request body is too large and exceeds the maximum permissible limit.
   Content-Length: [290]
   Content-Type: [application/xml]
   Date: [Sun, 14 Oct 2018 18:15:15 GMT]
   Server: [Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0]
   X-Ms-Error-Code: [RequestBodyTooLarge]
   X-Ms-Request-Id: [5d959ea2-101e-013b-68e9-6391bf000000]
   X-Ms-Version: [2018-03-28]

```
I tried chunk size 99MiB:
It runs for 5 minutes then I get:
<pre>
2018/10/14 13:25:52 INFO  : 
Transferred:   	      220M / 220 MBytes, 100%, 804.531 kBytes/s, ETA 0s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:       4m40s
Transferring:
 *                                      test.bin: 100% /220M, 1/s, 0s

2018/10/14 13:25:54 DEBUG : pacer: Rate limited, increasing sleep to 20ms
2018/10/14 13:25:54 DEBUG : pacer: low level retry 1/1 (error -> github.com/ncw/rclone/vendor/github.com/Azure/azure-pipeline-go/pipeline.newDefaultHTTPClientFactory.func1.1, /home/will/go/src/github.com/ncw/rclone/vendor/github.com/Azure/azure-pipeline-go/pipeline/core.go:234
HTTP request failed

Put https://XXX.blob.core.windows.net/test/test.bin?blockid=TfHODY8dT%2Fx12JewNOE6xwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA%3D%3D&comp=block&timeout=61: context deadline exceeded
)
2018/10/14 13:25:54 DEBUG : test.bin: Received error: -> github.com/ncw/rclone/vendor/github.com/Azure/azure-pipeline-go/pipeline.newDefaultHTTPClientFactory.func1.1, /home/will/go/src/github.com/ncw/rclone/vendor/github.com/Azure/azure-pipeline-go/pipeline/core.go:234
HTTP request failed

Put https://XXX.blob.core.windows.net/test/test.bin?blockid=TfHODY8dT%2Fx12JewNOE6xwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA%3D%3D&comp=block&timeout=61: context deadline exceeded
 - low level retry 1/10
2018/10/14 13:26:54 INFO  : 
Transferred:   	      220M / 440 MBytes, 50%, 658.732 kBytes/s, ETA 5m41s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:     5m41.9s
Transferring:
 *                                      test.bin:  0% /220M, 0/s, -
</pre>
I tested the blob API with curl and it respects timeout and the transfer fails exactly at the time limit. So the transfers are probably timing out here in this last test and the code is not detecting it?

**Comment by ncw** [comment]
@wpbrown - thanks for your analysis - it sounds like you are getting somewhere :-)

> I noticed in my log above the error was for comp=blocklist. But I know in the past that I've seen the error with comp=block. Neither of which makes sense because my blob is 220MiB, < 256MiB chunking cutoff. It looks like azblob.UploadStreamToBlockBlob used for the non-chunk rclone backend is always using the chunking REST API (see the methods of uploadStreamToBlockBlobOptions).

Yes, that was the conclusion I came to when investigating the problem that the `+1` you noticed fixed.

> This would still work, but its still including that timeout=61 even for the block PUTs. You might not notice this if you use the default 4MiB chunk size. However, I had set the chunk size very large, so with the 1min timeout, even the chunk puts are presumably timing out.

I haven't been able to figure out how to change or remove that timeout using the SDK - any ideas?

> With chunksize 100MiB:
> 
> ```
> BufferSize:      int(o.fs.opt.ChunkSize) + 1, // +1 Needed until https://github.com/Azure/azure-storage-blob-go/pull/75 is merged
> ```
> Also the +1 here is causing the valid chunksize config of 100MiB to fail:

I added the +1 in order to work around a problem where if you tried to upload a file of exactly the chunk size but below the upload cutoff it would fail.  I [found and fixed the problem](https://github.com/Azure/azure-storage-blob-go/pull/75) but I wasn't aware it would have that side effect.

I've made another issue #2653 to fix that in a better way.

> I tested the blob API with curl and it respects timeout and the transfer fails exactly at the time limit. So the transfers are probably timing out here in this last test and the code is not detecting it?

Yes the 5m timeout is suspiciously the default value for the IO idle timeout `--timeout` - you could try setting that to something different to prove that.

**Comment by ncw** [comment]
@wpbrown I made a much better fix for the SDK bug in 7f0b2042925b88c0f510df15026fdb1f7b5a8601 which should enable uploading with chunk size of 100M again.

https://beta.rclone.org/v1.43-173-g7f0b2042-beta/ (uploaded in 15-30 mins)

**Comment by ncw** [comment]
@wpbrown 

Great investigation - thanks.

If we want to change this, it looks related to the fix needed for #2647 - making a new Pipeline.

What do you think @sandeepkru ?

**Comment by sandeepkru** [comment]
@wpbrown - Agree, these are good points on retry policy behavior. Since rclone has generic pacer implementation around all calls, we did not change any pipe line options with SDK. As @ncw  mentioned we can consider exploring on this with #2654 fix.

**Comment by patrickli** [comment]
I am seeing the same problem downloading from Azure:

```
[ec2-user@ip-172-30-1-218 ~]$ rclone -P copy azure:consumers/20171021/consumers-20171021.zip .
2018-10-29 07:31:00 NOTICE: consumers-20171021.zip: Removing partially written file on error: context deadline exceeded
2018-10-29 07:32:00 NOTICE: consumers-20171021.zip: Removing partially written file on error: context deadline exceeded
2018-10-29 07:33:00 NOTICE: consumers-20171021.zip: Removing partially written file on error: context deadline exceeded
2018-10-29 07:34:00 NOTICE: consumers-20171021.zip: Removing partially written file on error: context deadline exceeded
2018-10-29 07:35:01 NOTICE: consumers-20171021.zip: Removing partially written file on error: context deadline exceeded
2018-10-29 07:36:01 NOTICE: consumers-20171021.zip: Removing partially written file on error: context deadline exceeded
Transferred:   	    1.250G / 1.862 GBytes, 67%, 3.369 MBytes/s, ETA 3m5s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:     6m19.9s
Transferring:
 *                        consumers-20171021.zip:  7% /675.108M, 3.200M/s, 3m15s^C
```

**Comment by brused27** [comment]
I was able to fix this and extend the generated timeout by setting the azblob Pipeline Options for "TryTimeout" but I have no idea what a reasonable number this should really be set to.  Azure seems to close the connection as soon as the total time the connection has been open meets the specified TryTimeout even if there is an active data transfer.  This is problematic on large files as it is not possible to pull them down quickly enough from Azure.

In my fix below, I'm setting these to 48 hours as I was pulling very large files down on a slow connection.  Perhaps this should be defaulted to a large value and override-able with configuration?

I updated this line to set the PipelineOptions:
https://github.com/ncw/rclone/blob/1f05d5bf4a515ba43dd59913fc83f87a7388e43b/backend/azureblob/azureblob.go#L325

and updated it to:
`pipeline := azblob.NewPipeline(credential, azblob.PipelineOptions{Retry: azblob.RetryOptions{ TryTimeout: time.Hour * 48 }})`

I also updated this line to set the PipelineOptions:
https://github.com/ncw/rclone/blob/1f05d5bf4a515ba43dd59913fc83f87a7388e43b/backend/azureblob/azureblob.go#L334

and updated it to:
`pipeline := azblob.NewPipeline(azblob.NewAnonymousCredential(), azblob.PipelineOptions{Retry: azblob.RetryOptions{ TryTimeout: time.Hour * 48 }})`


See the Azure storage blob for Go SDK reference here:

[TryTimeout Reference](https://github.com/Azure/azure-storage-blob-go/blob/5152f14ace1c6db66bd9cb57840703a8358fa7bc/azblob/zc_policy_retry.go#L39-L44)

I would be happy to submit a pull request if anyone has a suggestion on how to set the TryTimeout to an acceptable number as the documents suggest.

**Comment by ncw** [comment]
@brused27 wrote:
> I would be happy to submit a pull request if anyone has a suggestion on how to set the TryTimeout to an acceptable number as the documents suggest.

Great!

I'd be happy if the TryTimeout is set very large (effectively infinite) as rclone implements an idle timeout in the http transport anyway.   An idle timeout is a much better fit for file transfers than a fixed timeout.

 (Actually the custom http transport isn't currently working for azureblob but it should be fixed shortly in #2654)

This could easily be an advanced parameter to the backend if we wanted.

**Comment by ccvalley** [comment]
Was the solution merged to the beta version? Or which version of rclone should I be using to avoid this issue?
Thanks.
